src/test/java/com/mongodb/hibernate/jdbc/MongoResultSetTests.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.mongodb.client.MongoCursor;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.bson.BsonDocument;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MongoResultSetTests {
+    private static final List<String> FIELDS = List.of("id", "title", "author");
+
+    @Mock
+    private MongoCursor<BsonDocument> mongoCursor;
+
+    private MongoResultSet mongoResultSet;
+
+    @BeforeEach
+    void setUp() {
+        mongoResultSet = new MongoResultSet(mongoCursor, FIELDS);
+    }
+
+    @Nested
+    class CloseTests {
+
+        @BeforeEach
+        void setUp() throws SQLException {
+            mongoResultSet.close();
```

Comment:
It seems this is the "when", so it should be inside the test? I think it is fine to repeat it.

Reply:
This is not "when", it is a "given". All the testing cases are focusing on what would happen when we do othing thing (re-close it, invoking methods on it, etc.) is based on such pre-condition. For that reason, it aligns with the intention of `@BeforeEach`.

Reply:
That makes sense. However, this `@Nested` is called "Close", so (based on how these were grouped, according to method) I would expect it to be testing the close method. I am not sure I understand when nested groupings should be used, or what I should expect to find in them.

Reply:
This is a well-established convention applying to the testing code for `Statement`, `Connection` and `PreparedStatement`. It means we test the behaviour when the closable object has been closed (usually including re-close ending up with no-op, SQLException being thrown when any method is invoked).

Sorry for the confusion. Welcome to any naming suggestion, which could be applied to all the previous JDBC closing testing cases.